cypress/e2e/gui-tests/customers.cy.js

```diff
@@ -0,0 +1,202 @@
+
+    sizes.forEach((size, index) => {
+      cy.request('GET', `${CUSTOMERS_API}?size=${size}`).as('getSizedCustomers')
+
+      cy.get('@getSizedCustomers')
+        .its('body.customers')
+        .each(customer => {
+          expect(customer.size).to.eq(size)
+          expect(customer.employees).to.be.lte(limitOfEmployessPerSize[index])
+        })
+    })
+  })
+
+  it('filters by All sizes', () => {
+    cy.intercept(
+      'GET',
+      `${Cypress.env('API_URL')}/customers?page=1&limit=10&size=All`,
+      { fixture: 'allCustomers' }
+    ).as('getAllCustomers')
+
+    cy.get('[data-testid="filter"]').select('All')
+    cy.wait('@getAllCustomers')
+
+    cy.get('tbody tr').should('have.length', 10)
+  })
+
+  it('filters by Small size', () => {
+    cy.intercept(
+      'GET',
+      `${Cypress.env('API_URL')}/customers?page=1&limit=10&size=Small`,
+      { fixture: 'smallCustomers' }
+    ).as('getSmallCustomers')
+
+    cy.get('[data-testid="filter"]').select('Small')
+
+    cy.wait('@getSmallCustomers')
+
+    cy.get('tbody tr').should('have.length', 5)
+  })
+
+  it('filters by Medium size', () => {
+    cy.intercept(
+      'GET',
+      `${Cypress.env('API_URL')}/customers?page=1&limit=10&size=Medium`,
+      { fixture: 'mediumCustomers' }
+    ).as('getMediumCustomers')
+
+    cy.get('[data-testid="filter"]').select('Medium')
+
+    cy.wait('@getMediumCustomers')
+
+    cy.get('tbody tr').should('have.length', 5)
+  })
+
+  it('filters by Enterprise size', () => {
+    cy.intercept(
+      'GET',
+      `${Cypress.env('API_URL')}/customers?page=1&limit=10&size=Enterprise`,
+      { fixture: 'enterpriseCustomers' }
+    ).as('getEnterpriseCustomers')
+
+    cy.get('[data-testid="filter"]').select('Enterprise')
+
+    cy.wait('@getEnterpriseCustomers')
+
+    cy.get('tbody tr').should('have.length', 1)
+  })
+
+  it('filters by Large Enterprise size', () => {
+    cy.intercept(
+      'GET',
+      `${Cypress.env('API_URL')}/customers?page=1&limit=10&size=Large%20Enterprise`,
+      { fixture: 'largeEnterpriseCustomers' }
+    ).as('getLargeEnterpriseCustomers')
+
+    cy.get('[data-testid="filter"]').select('Large Enterprise')
+
+    cy.wait('@getLargeEnterpriseCustomers')
+
+    cy.get('tbody tr').should('have.length', 3)
+  })
+
+  it('filters by Very Large Enterprise size', () => {
+    cy.intercept(
+      'GET',
+      `${Cypress.env('API_URL')}/customers?page=1&limit=10&size=Very%20Large%20Enterprise`,
+      { fixture: 'veryLargeEnterpriseCustomers' }
+    ).as('getVeryLargeEnterpriseCustomers')
+
+    cy.get('[data-testid="filter"]').select('Very Large Enterprise')
+
+    cy.wait('@getVeryLargeEnterpriseCustomers')
+
+    cy.get('tbody tr').should('have.length', 9)
+  })
+
+  it('Correctly downloads a list of customers as a CSV file', () => {
+    cy.readFile('cypress/downloads/customers.csv').should('not.exist')
+    cy.get('.download-csv-button').click()
```

Comment:
Que tal algo mais descritivo?

Assim:

```suggestion
    cy.contains('button', 'Download CSV').click()
```